ItemInvUtil#move fails if the target inventory does not accept the item

I am using:

`ItemInvUtil.move(extractable, insertable, 1)`

If the target insertable does not accept the item, the method crashes because it tries to extract 0 items from extractable, then complains that it gets an empty stack.

```
Caused by: java.lang.IllegalStateException: Tried to extract the filter (C) from A but it returned an empty item stack after we have already inserted the expected stack into B!
The inventory is now in an invalid (duped) state!
```

--- a/src/main/java/alexiil/mc/lib/attributes/item/ItemInvUtil.java
+++ b/src/main/java/alexiil/mc/lib/attributes/item/ItemInvUtil.java
@@ -110,6 +110,9 @@ public final class ItemInvUtil {
         }
         ItemStack leftover = to.attemptInsertion(extracted, simulation);
         int insertedAmount = extracted.getCount() - (leftover.isEmpty() ? 0 : leftover.getCount());
+        if (insertedAmount == 0) {
+            return 0; // Nothing was accepted by the target
+        }
         ItemStack reallyExtracted = from.attemptExtraction(insertionFilter, insertedAmount, simulation);
 
         if (reallyExtracted.isEmpty()) {

